dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -432,6 +432,426 @@ bool TypeLookupService::complete_to_minimal_type_object(const TypeObject& cto, T
+    } else if (cto.union_type.union_flags & (1 << 2)) {
+      dt->descriptor_->extensibility_kind = MUTABLE;
+    } else {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("Invalid extensibility kind in complete_to_dynamic(DynamicType& dt, const CompleteTypeObject& cto)\n")));
+    }
+    dt->descriptor_->is_nested = (cto.union_type.union_flags & (1 << 3));
+    for (ACE_CDR::ULong i = 0; i < cto.union_type.member_seq.length(); ++i) {
+      DynamicTypeMember_rch dtm(new DynamicTypeMember, OpenDDS::DCPS::keep_count());
+      MemberDescriptor* md(new MemberDescriptor);
+      dtm->descriptor_ = md;
+      complete_union_member_to_member_descriptor(dtm->descriptor_, cto.union_type.member_seq[i], dt_map);
+      dtm->descriptor_->index = i;
+      insert_dynamic_member(dt, dtm);
+    }
+    break;
+  case TK_BITSET:
+    dt->descriptor_->kind = TK_BITSET;
+    dt->descriptor_->name = cto.bitset_type.header.detail.type_name;
+    dt->descriptor_->bound.length(0);
+    break;
+  // Collection TKs
+  case TK_SEQUENCE:
+    dt->descriptor_->kind = TK_SEQUENCE;
+    if(cto.sequence_type.header.detail.present == true) {
+      dt->descriptor_->name = cto.sequence_type.header.detail.value.type_name;
+    } else {
+      dt->descriptor_->name = "";
+    }
+    dt->descriptor_->bound.length(1);
+    dt->descriptor_->bound[0] = cto.sequence_type.header.common.bound;
+    type_identifier_to_dynamic(dt->descriptor_->element_type, cto.sequence_type.element.common.type, dt_map);
+    break;
+  case TK_ARRAY: {
+    dt->descriptor_->kind = TK_ARRAY;
+    dt->descriptor_->name = cto.array_type.header.detail.type_name;
+    dt->descriptor_->bound = cto.array_type.header.common.bound_seq;
+    type_identifier_to_dynamic(dt->descriptor_->element_type, cto.array_type.element.common.type, dt_map);
+  }
+  break;
+  case TK_MAP:
+    dt->descriptor_->kind = TK_MAP;
+    if (cto.map_type.header.detail.present == true) {
+      dt->descriptor_->name = cto.map_type.header.detail.value.type_name;
+    } else {
+      dt->descriptor_->name = "";
+    }
+    dt->descriptor_->bound.length(1);
+    dt->descriptor_->bound[0] = cto.map_type.header.common.bound;
+    type_identifier_to_dynamic(dt->descriptor_->element_type, cto.map_type.element.common.type, dt_map);
+    type_identifier_to_dynamic(dt->descriptor_->key_element_type, cto.map_type.key.common.type, dt_map);
+    break;
+  }
+}
+
+void TypeLookupService::type_identifier_to_dynamic(DynamicType_rch& dt,
+  const TypeIdentifier& ti, DynamicTypeMap& dt_map)
+{
+  if (ti.kind() == TK_NONE) {
+    // Leave as dt as nil
+    return;
+  }
+  DynamicTypeMap::iterator ti_found = dt_map.find(ti);
+  if (ti_found != dt_map.end()) {
+    dt = ti_found->second;
+    return;
+  }
+  DynamicType_rch dt_instantiation(new DynamicType, OpenDDS::DCPS::keep_count());
+  dt = dt_instantiation;
+  dt_map.insert(dt_map.end(), std::make_pair(ti , dt));
+  TypeDescriptor* td(new TypeDescriptor);
+  dt->descriptor_ = td;
+    switch (ti.kind()) {
```

Comment:
```suggestion

    switch (ti.kind()) {
```
This `switch` block's indentation is off by 2 spaces.